langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -69,8 +70,10 @@ public T build() {
                 throw illegalConfiguration("The @Moderate annotation is present, but the moderationModel is not set up. " +
                         "Please ensure a valid moderationModel is configured before using the @Moderate annotation.");
             }
-            if (method.getReturnType() == Result.class) {
-                validateResultReturnType(method);
+            if (method.getReturnType() == Result.class ||
+                method.getReturnType() == List.class ||
```

Comment:
I guess the validation error message for `List` and `Set` without concrete types should mention `List` and `Set` instead of `Result`

Reply:
Right.. I've made these dynamically generated which should improve the UX (for the end users). Also moved few more type related methods from  `AiServices` to `TypeUtils` and covered everything with unit tests.